src/library/autodj/autodjfeature.cpp

```diff
@@ -298,32 +263,13 @@ void AutoDJFeature::slotAddRandomTrack(bool) {
 }
 
 void AutoDJFeature::constructCrateChildModel() {
-    // Create a crate table-model with a list of crates that have been added
-    // to the auto-DJ queue (and are visible).
-    QSqlTableModel crateListTableModel(this, m_pTrackCollection->getDatabase());
-    crateListTableModel.setTable(CRATE_TABLE);
-    crateListTableModel.setSort(crateListTableModel.fieldIndex(CRATETABLE_NAME),
-                                Qt::AscendingOrder);
-    crateListTableModel.setFilter(CRATETABLE_AUTODJ_SOURCE + " = 1 AND " + CRATETABLE_SHOW + " = 1");
-    crateListTableModel.select();
-    while (crateListTableModel.canFetchMore()) {
-        crateListTableModel.fetchMore();
-    }
-
-    QSqlRecord tableModelRecord = crateListTableModel.record();
-    int nameColumn = tableModelRecord.indexOf(CRATETABLE_NAME);
-    int idColumn = tableModelRecord.indexOf(CRATETABLE_ID);
-
-    // Create a tree-item for each auto-DJ crate.
-    for (int row = 0; row < crateListTableModel.rowCount(); ++row) {
-        int id = crateListTableModel.data(
-            crateListTableModel.index(row, idColumn)).toInt();
-        QString name = crateListTableModel.data(
-            crateListTableModel.index(row, nameColumn)).toString();
-        m_crateList.append(qMakePair(id, name));
-
+    m_crateList.clear();
+    CrateSelectIterator autoDjCrates(m_pTrackCollection->crates().selectAutoDjCrates(true));
+    Crate crate;
+    while (autoDjCrates.readNext(&crate)) {
         // Create the TreeItem for this crate.
-        m_pCratesTreeItem->appendChild(name);
+        m_pCratesTreeItem->appendChild(crate.getName(), crate.getId().toVariant());
+        m_crateList.append(std::move(crate));
```

Comment:
QList will never allow move semantics. I have implement move semantics since it hat the copy on write feature. 
If you desire move, please use std container, else remove the std::move since it is IMHO unnecessary clutter.